### PR TITLE
Add support for int8/16 to parquet encode

### PR DESCRIPTION
**Pull request by richardartoul**

### CODE REVIEW

File: internal/impl/parquet/convert.go [hunk 1]
```diff
@@ -99,44 +197,65 @@ func setField(field reflect.Value, value any) error {
...
 			field.SetFloat(float64(v))
-			return nil
+		case int8:
+			field.SetFloat(float64(v))
+		case int16:
+			field.SetFloat(float64(v))
+		case int32:
+			field.SetFloat(float64(v))
 		case int64:
 			field.SetFloat(float64(v))
```
Comment (arjunnair1997) [inline internal/impl/parquet/convert.go:254]: is int64 to float 32 lossy?

File: internal/impl/parquet/convert.go [hunk 2]
```diff
@@ -99,44 +197,65 @@ func setField(field reflect.Value, value any) error {
...
-			return nil
+		case float32:
+			field.SetFloat(float64(v))
+		case float64:
+			field.SetFloat(v)
```
Comment (arjunnair1997) [inline internal/impl/parquet/convert.go:258]: Is this lossy? Or is the float32 field just stored as float64

**Review by arjunnair1997 — APPROVED** [review]

File: internal/impl/parquet/convert.go (hunk 1 above)
Comment (richardartoul) [reply internal/impl/parquet/convert.go:254]: definitely lossy, but wont underflow/overflow

File: internal/impl/parquet/convert.go (hunk 2 above)
Comment (richardartoul) [reply internal/impl/parquet/convert.go:258]: I think so, but would have to be a humongous number. We can fix this later if it comes up, thats how it worked before

**Pull request merged**